functions/populateCounselors.ts

```diff
@@ -0,0 +1,54 @@
+import '@twilio-labs/serverless-runtime-types';
+import {
+  Context,
+  ServerlessCallback,
+  ServerlessFunctionSignature,
+} from '@twilio-labs/serverless-runtime-types/types';
+
+const TokenValidator = require('twilio-flex-token-validator').functionValidator;
+
+type EventBody = {
+  workspaceSID: string | undefined;
+};
+
+export const handler: ServerlessFunctionSignature = TokenValidator(
+  async (context: Context, event: {}, callback: ServerlessCallback) => {
+    const response = new Twilio.Response();
+    response.appendHeader('Access-Control-Allow-Origin', '*');
+    response.appendHeader('Access-Control-Allow-Methods', 'OPTIONS, POST, GET');
+    response.appendHeader('Access-Control-Allow-Headers', 'Content-Type');
+
+    try {
+      const body = event as EventBody;
+      const { workspaceSID } = body;
+
+      if (workspaceSID === undefined) {
+        throw new Error('Error: Workspace parameter not provided');
+      }
+
+      const workspace = await context
+        .getTwilioClient()
+        .taskrouter.workspaces(workspaceSID)
+        .fetch();
+
+      const workers = await workspace.workers().list();
+
+      const prettyWorkers = workers.map(w => ({
+        sid: w.sid,
+        friendlyName: w.friendlyName,
+      }));
+
+      response.setStatusCode(200);
+      response.appendHeader('Content-Type', 'application/json');
+      response.setBody({ prettyWorkers });
+      callback(null, response);
+    } catch (err) {
```

Comment:
This error handling needs to be better.  When I send a request without a `workspaceSID`, I get back a 500 and a body of `{}`.  We should get back a 400 (because the request is malformed) and get back the error message in the body.

In the case where the call to the Worker resource fails, we should return a 502 (because an upstream service failed) and _probably_ return the original error from the Worker resource in the body.